contessa/runner.py

```diff
@@ -27,24 +27,32 @@ def __init__(self, conn_uri_or_engine, special_qc_map=None):
     def run(
         self,
         raw_rules: List[Dict[str, str]],
-        table_name: str,
-        schema_name: str,
-        task_time: datetime = None,
+        check_table: Dict,
+        result_table: Dict,  # todo - docs for quality name, maybe defaults..
+        context: Optional[Dict] = None,
     ):
+        check_table = Table(**check_table)
+        result_table = QualityTable(**result_table)
+
         normalized_rules = self.normalize_rules(raw_rules)
-        refresh_executors(schema_name, table_name, self.conn, task_time)
-        quality_check_class = self.get_quality_check_class(table_name)
+        refresh_executors(check_table, self.conn, context)
+        quality_check_class = self.get_quality_check_class(result_table)
+
+        # update schema name where user want to put the dq check results
+        # todo - test this
+        quality_check_class.metadata.schema = result_table.schema_name
         self.ensure_table(quality_check_class)
 
         rules = self.build_rules(normalized_rules)
-        objs = self.do_quality_check(quality_check_class, rules, task_time)
+        objs = self.do_quality_check(quality_check_class, rules, context)
 
         self.insert(objs)
 
     def normalize_rules(self, raw_rules):
         return RuleNormalizer().normalize(raw_rules)
 
-    def do_quality_check(self, dq_cls, rules: List[Rule], task_time: datetime = None):
+    # todo - solve task time
+    def do_quality_check(self, dq_cls, rules: List[Rule], context: Dict = None):
```

Comment:
Perhaps it would make sense to make the `do_quality_check` method atomic, i.e. perform the check on one rule instead a list of rules - just to make it a bit simpler and easier to test - what do you think?

Reply:
changed